src/librustc_typeck/lib.rs

```diff
@@ -340,6 +341,8 @@ pub fn provide(providers: &mut Providers) {
 pub fn check_crate<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>)
                              -> Result<(), CompileIncomplete>
 {
+    tcx.sess.profiler(|p| p.start_activity(ProfileCategory::TypeChecking));
```

Comment:
(is there a non-trivial amount of work here that takes place outside of queries?)

Reply:
Hmm... I'm not sure off hand. I can investigate.